### DISCUSSION TITLE
[BP#13] Favoriser les pages statiques

### DISCUSSION THREAD

**Discussion by devcrafting** [cnumr/best-practices#206] [Bonnes pratiques] [answered]
Finalement cette BP est aussi redondante avec la [BP 15](https://github.com/cnumr/best-practices/blob/main/chapters/BP_015_fr.md), c'est le premier type de techno de la liste, voire un peu du 2.

Je ne connais aucune des stacks proposées, je ne sais pas dire si c'est encore d'actualité.

Enfin la métrique me paraît compliquée à appliquer, le % de pages statiques étant très variable en fonction du besoin.

**Comment by ACTLEM** [comment]
Les stacks proposées sont pour moi encore d'actualités (du moins elles ne me semblent pas encore dépassées).

Je pense que c'est bien d'avoir une BP propre pour insister sur les pages statiques quitte à revoir la BP 15 pour éviter d'avoir une redondance.

Concernant la métrique, je suis d'accord cela dépend beaucoup du besoin...mais j'avoue que c'est compliqué de trouver une règle simple.

**Reply by devcrafting** [reply]
comme proposé dans la [BP 15](https://github.com/cnumr/best-practices/blob/main/chapters/BP_015_fr.md), peut-être plus favoriser 5 BPs (dont celle-ci) plutôt que noyer cela dans une BP générale ?

**Reply by devcrafting** [reply]
Et du coup pour celle-ci, la mesure serait plus sur les critères de choix d'une architecture à base de pages statiques

**Comment by Fred-artwai** [comment]
Alors, je trouve cette BP intéressante. Elle a le mérite de bien préciser l'avantage d'un fichier statique en terme d'écoconception.
On peut avoir un site dynamique géré par un CMS et distribuer les pages de manière statique or ce cas n'est pas décris dans la BP 15. Exemple WordPress avec un plugin de cache comme WP Rocket ou Total Cache. Vu que 40% des sites de la planète utilise WordPress cet exemple pourrait avoir plus d'impact que Jamstack.

Reformulation "Préférer l'envoi de ressources statiques"
Que cela soit un site nativement statique ou des fichiers statiques générés par un CMS ou autre.

**Reply by devcrafting** [reply]
oui dans ce cas, cela irait clairement dans une BP dédiée au choix d'une solution technologie type CMS (j'ai proposé qu'on découpe en BP les choix technologiques de la 15).
a minima, il faudrait mentionner ce que tu dis dans la BP :)

**Reply by tbroyer** [reply]
Ces plugins me font plus penser à la BP#57.

Cette BP ci est plus orientée sur un choix d'architecture: un "site plaquette" n'a aucun intérêt à être propulsé par WordPress, même avec toutes les mises en cache du monde.
Autre exemple: la première version d'EcoIndex était basée sur WordPress pour 3 pages qui ne changent jamais (accueil, qu'est-ce que ecoindex, à propos), une bonne pratique en page d'accueil servie dynamiquement (random côté serveur) et chargée en AJAX, et des pages dynamiques d'attente puis de résultats d'analyses; ces 3 pages qui ne changent jamais auraient dû être statiques (ce sera visiblement le cas sur la v2), et le random de la bonne pratique aurait pu être fait côté client pour aller chercher une ressource statique également, et le reste n'à rien à voir avec du CMS et a été développé sur-mesure.

**Comment by ACTLEM** [comment]
En attente décision https://github.com/cnumr/best-practices/discussions/199, même si elle devrait rester (avec peut-être du rewording).

**Comment by ACTLEM** [comment, marked as answer]
Cette BP a un intérêt (la BP sera au minimum retravaillé). Si quelqu'un souhaite l'améliorer qu'il n'hésite pas à proposer une Pull Request. Pour l'instant, je n'ai pas créé d'issue pour la laisser telle quelle.

**Reply by florinesueur** [reply]
Je pense qu'on peut lui apporter des exemples concrets, plus tournés vers le choix de technologie de la BP15.

**Reply by ACTLEM** [reply]
Ca marche, j'ai créé un issue pour modifications des exemples: https://github.com/cnumr/best-practices/issues/333
